tls/s2n_client_cert.c

```diff
@@ -27,6 +27,71 @@
 #include "utils/s2n_blob.h"
 #include "utils/s2n_safety.h"
 
+/* In TLS1.2, the certificate list is just an opaque vector of certificates:
+ *
+ *      opaque ASN.1Cert<1..2^24-1>;
```

Comment:
Nit: room for a compliance comment here?

Reply:
I'm not sure it'd be helpful / correct to add one. This code isn't really enforcing any part of the RFC; it's just converting the new format to the old format so the old method can work the same in TLS1.2 and TLS1.3.